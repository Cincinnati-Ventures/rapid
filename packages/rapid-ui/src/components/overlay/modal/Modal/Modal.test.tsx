import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import {
	Modal,
	ModalOverlay,
	ModalContent,
	ModalDescription,
	ModalTitle,
} from '../';

describe('Modal', () => {
	it('should render default correctly', () => {
		const { container } = render(
			<Modal open={true} onClose={() => {}}>
				<ModalOverlay />
				<ModalContent styles='flex flex-col'>
					<ModalTitle>Delete Account</ModalTitle>

					<p>
						Are you sure you want to delete your account? All of
						your data will be permanently removed. This action
						cannot be undone.
					</p>
					<div className='self-end'></div>
				</ModalContent>
			</Modal>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(
			<Modal styles='bg-green' open={true} onClose={() => {}} />,
		);
		expect(container).toMatchSnapshot();
	});
});
