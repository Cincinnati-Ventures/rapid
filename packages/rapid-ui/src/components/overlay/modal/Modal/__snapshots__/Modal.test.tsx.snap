// Vitest Snapshot v1

exports[`Modal > should render correctly with non default props 1`] = `
<div
  aria-hidden="true"
>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0; padding: 0; margin: -1px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0; display: none;"
  />
</div>
`;

exports[`Modal > should render default correctly 1`] = `
<div
  aria-hidden="true"
>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0; padding: 0; margin: -1px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0; display: none;"
  />
</div>
`;
